woodwork/tests/type_system/test_ltype_inference.py

```diff
@@ -120,7 +120,9 @@ def test_categorical_inference(categories):
     dtypes = ["object", "string", "category"]
     if _is_spark_series(categories[0]):
         dtypes = get_spark_dtypes(dtypes)
-    for series in categories:
+    for ind, series in enumerate(categories):
+        if ind == len(categories) - 1:
+            dtypes = ["string", "category"]
```

Comment:
Casting an all integer series as an `object` and then attempting to have it be inferred as categorical will not work anymore, as the nan replacement will make it an `int` dtype. This will still work for `string` and `categorical` casting.